core/remotes/docker/auth/fetch.go

```diff
@@ -95,6 +96,10 @@ type OAuthTokenResponse struct {
 
 // FetchTokenWithOAuth fetches a token using a POST request
 func FetchTokenWithOAuth(ctx context.Context, client *http.Client, headers http.Header, clientID string, to TokenOptions) (*OAuthTokenResponse, error) {
+	c := *client
+	client = &c
+	tracing.UpdateHTTPClient(client, tracing.Name("remotes.docker.resolver", "FetchTokenWithOAuth"))
```

Comment:
Not 100% sure what the conventions are; should this include "auth" (`remotes.docker.auth.resolver`) if this follows the name of the package we're in?